public/components/layer_control/layer_configuration_options.tsx

```diff
@@ -42,13 +47,13 @@ export type MapsExlorerOptionsProps = VisOptionsProps<MapsExplorerVisParams> & {
 function LayerConfigurationOptions(props: MapsExlorerOptionsProps) {
   const { stateParams, setValue, vis, setValidity, configLayerId, configMode } = props;
   // specific layer's input options validity
-  const [optionValidity, setOptionValidity] = useState(true); 
+  const [optionValidity, setOptionValidity] = useState(true);
 
   //update layer's options
   const setLayerValue = <T extends keyof LayerOptions>(paramName: T, value: LayerOptions[T]) => {
     // during the creatation, when users switch layer type from other types to TMS, 
-    //set optionValidity Trur to ensure users are able to creat new TMS layer
-    if (paramName === "layerType") {setOptionValidity(true);};
+    //set optionValidity True to ensure users are able to creat new TMS layer
+    if (paramName === "layerType") { setOptionValidity(true); };
```

Comment:
what all values can paramName have? the check can be reversed and a constant can be used in place of the string.

Reply:
- paramName could be any keys in LayerOptions, such as id, name, layerType, typeOptions, minZoom, maxZoon, isHidden..
- Gonna reverse and replace it